packages/multichain-transactions-controller/src/MultichainTransactionsController.ts

```diff
@@ -381,6 +415,7 @@ export class MultichainTransactionsController extends BaseController<
 
         filteredNewTransactions.forEach((tx) => {
           transactions.set(tx.id, tx);
+          this.#publishTransactionUpdateEvent(tx);
```

Comment:
I think that events should come after the state really got updated? IDK if that's something we enforce, but if you trigger some logic for some events and trying to pick some other data on your state, you might have different values (mainly if you don't use the data that comes with the event)